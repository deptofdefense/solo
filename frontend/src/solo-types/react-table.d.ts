import {
  UseExpandedHooks,
  UseExpandedInstanceProps,
  UseExpandedOptions,
  UseExpandedRowProps,
  UseExpandedState,
  UseFiltersColumnOptions,
  UseFiltersColumnProps,
  UseFiltersInstanceProps,
  UseFiltersOptions,
  UseFiltersState,
  UseGlobalFiltersInstanceProps,
  UseGlobalFiltersOptions,
  UseGlobalFiltersState,
  UsePaginationInstanceProps,
  UsePaginationOptions,
  UsePaginationState,
  UseRowStateCellProps,
  UseRowStateInstanceProps,
  UseRowStateOptions,
  UseRowStateRowProps,
  UseRowStateState,
  UseSortByColumnOptions,
  UseSortByColumnProps,
  UseSortByHooks,
  UseSortByInstanceProps,
  UseSortByOptions,
  UseSortByState
} from "react-table";

declare module "react-table" {
  export interface TableOptions<D extends object>
    extends UseExpandedOptions<D>,
      UseFiltersOptions<D>,
      UseGlobalFiltersOptions<D>,
      UsePaginationOptions<D>,
      UseRowStateOptions<D>,
      UseSortByOptions<D>,
      Record<string, any> {}

  export interface Hooks<D extends object = {}>
    extends UseExpandedHooks<D>,
      UseSortByHooks<D> {}

  export interface TableInstance<D extends object = {}>
    extends UseColumnOrderInstanceProps<D>,
      UseExpandedInstanceProps<D>,
      UseFiltersInstanceProps<D>,
      UseGlobalFiltersInstanceProps<D>,
      UsePaginationInstanceProps<D>,
      UseRowStateInstanceProps<D>,
      UseSortByInstanceProps<D> {}

  export interface TableState<D extends object = {}>
    extends UseColumnOrderState<D>,
      UseExpandedState<D>,
      UseFiltersState<D>,
      UseGlobalFiltersState<D>,
      UsePaginationState<D>,
      UseRowStateState<D>,
      UseSortByState<D> {}

  export interface Column<D extends object = {}>
    extends UseFiltersColumnOptions<D>,
      UseSortByColumnOptions<D> {}

  export interface ColumnInstance<D extends object = {}>
    extends UseFiltersColumnProps<D>,
      UseSortByColumnProps<D> {}

  export interface Cell<D extends object = {}>
    extends UseRowStateCellProps<D> {}

  export interface Row<D extends object = {}>
    extends UseExpandedRowProps<D>,
      UseRowStateRowProps<D> {}
}
