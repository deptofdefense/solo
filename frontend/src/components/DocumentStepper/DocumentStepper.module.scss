.stepRoot {
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  align-content: space-around;
  margin-top: 1.8rem;
}

.step {
  width: 100%;
  display: flex;
  position: relative;
  flex-direction: column;
  align-items: center;
}

.stepIcon {
  z-index: 20;
  background-color: white;
  padding-left: 2px;
  padding-right: 2px;
  font-size: 1.8rem;
}

.stepComplete {
  color: green;
}

.stepIncomplete {
  color: lightgray;
}

.stepRight {
  position: absolute;
  height: 2px;
  background-color: lightgray;
  width: 50%;
  left: 50%;
  top: 0.9rem;
}

.stepLeft {
  position: absolute;
  height: 2px;
  background-color: lightgray;
  width: 50%;
  right: 50%;
  top: 0.9rem;
}