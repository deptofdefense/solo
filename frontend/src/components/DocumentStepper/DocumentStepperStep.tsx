import React from "react";
import classNames from "classnames";
import { formatDistanceToNow, parseISO } from "date-fns";
import {
  FontAwesomeIcon,
  FontAwesomeIconProps
} from "@fortawesome/react-fontawesome";
import { faCheckCircle, faDotCircle } from "@fortawesome/free-solid-svg-icons";
import classes from "./DocumentStepper.module.scss";

interface IconProps extends Omit<FontAwesomeIconProps, "icon"> {
  complete?: boolean;
}

const StatusIcon: React.FC<IconProps> = ({ complete, ...rest }) => (
  <FontAwesomeIcon {...rest} icon={complete ? faCheckCircle : faDotCircle} />
);

interface StepProps {
  title: string;
  first?: boolean;
  last?: boolean;
  occuredAt?: string;
  complete?: boolean | string | undefined;
}

const Step: React.FC<StepProps> = ({
  first = false,
  last = false,
  complete = false,
  title,
  occuredAt
}) => (
  <div
    className={classNames(classes.step, {
      [classes.stepComplete]: complete,
      [classes.stepIncomplete]: !complete
    })}
  >
    <StatusIcon className={classes.stepIcon} complete={Boolean(complete)} />
    <div>{title}</div>
    {occuredAt && (
      <div>{`${formatDistanceToNow(parseISO(occuredAt))} ago `}</div>
    )}
    {!first && <div className={classes.stepLeft} />}
    {!last && <div className={classes.stepRight} />}
  </div>
);

export default Step;
