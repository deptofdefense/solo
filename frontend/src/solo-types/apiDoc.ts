import { ApiDocument, Status } from "solo-types";
import * as faker from "faker";

export const defaultApiDoc: ApiDocument = {
  id: 1,
  statuses: [
    {
      id: 1,
      dic: {
        id: 1,
        code: "AE1",
        desc: "in transit"
      },
      status_date: "2020-03-01T21:47:13-05:00",
      key_and_transmit_date: null,
      esd: "2020-03-20",
      projected_qty: 2,
      received_qty: null,
      document: 1
    },
    {
      id: 2,
      dic: {
        id: 2,
        code: "AS1",
        desc: "received"
      },
      status_date: "2020-03-05T21:47:56-05:00",
      key_and_transmit_date: null,
      esd: null,
      projected_qty: 2,
      received_qty: null,
      document: 1
    }
  ],
  suppadd: {
    id: 1,
    subinventorys: [
      {
        id: 1,
        locators: [
          {
            id: 1,
            code: "M1234AA",
            desc: "",
            subinventorys: 1
          },
          {
            id: 2,
            code: "M1235A",
            desc: "",
            subinventorys: 1
          },
          {
            id: 3,
            code: "M1236",
            desc: "",
            subinventorys: 1
          }
        ],
        code: "MTM_STGE",
        desc: "",
        suppadd: 1
      },
      {
        id: 2,
        locators: [],
        code: "MTM_LAY",
        desc: "",
        suppadd: 1
      },
      {
        id: 3,
        locators: [],
        code: "MTM_PEB",
        desc: "",
        suppadd: 1
      }
    ],
    code: "YMTM",
    desc: "Motor Transportation"
  },
  part: {
    id: 2,
    nsn: "5430015061999",
    nomen: "TANK,LIQUID STORAGE",
    uom: "ea",
    price: 15,
    sac: 1,
    serial_control_flag: "n",
    lot_control_flag: "n",
    recoverability_code: "z",
    shelf_life_code: 0,
    controlled_inv_item_code: "u"
  },
  service_request: {
    id: 1,
    service_request: "12345678"
  },
  addresses: [
    {
      id: 4,
      address_type: {
        id: 4,
        type: "Bill-To",
        desc: null
      },
      name: "AAC-M30300",
      ric: "SMS",
      addy1: "addy1",
      addy2: "addy2",
      addy3: "bldg 24006 Montezuma",
      city: "Arlington",
      state: "VA",
      zip: "22202",
      country: "United States",
      document: [1]
    },
    {
      id: 3,
      address_type: {
        id: 3,
        type: "Requester",
        desc: null
      },
      name: "requestor",
      ric: "5E3",
      addy1: "addy1",
      addy2: "addy2",
      addy3: "addy3",
      city: "Arlington",
      state: "VA",
      zip: "22202",
      country: "United States",
      document: [1]
    },
    {
      id: 2,
      address_type: {
        id: 2,
        type: "Ship-To",
        desc: null
      },
      name: "ship to",
      ric: "SMS",
      addy1: "addy1",
      addy2: "addy2",
      addy3: "addy3",
      city: "Arlington",
      state: "VA",
      zip: "22202",
      country: "United States",
      document: [1]
    },
    {
      id: 1,
      address_type: {
        id: 1,
        type: "Holder",
        desc: null
      },
      name: "AAC-M30300",
      ric: "SMS",
      addy1: "addy1",
      addy2: "addy2",
      addy3: "addy3",
      city: "Arlington",
      state: "VA",
      zip: "22202",
      country: "United States",
      document: [1]
    }
  ],
  sdn: "M3030012341234"
};

const idxToStatus = (idx: number): Status => {
  const base: Omit<Status, "dic"> = {
    id: Math.floor(Math.random() * 20),
    status_date: faker.date.recent().toISOString(),
    key_and_transmit_date: null,
    esd: "2020-03-20",
    projected_qty: 20,
    received_qty: faker.random.number(20),
    document: 1
  };
  switch (idx) {
    case 0:
      return {
        ...base,
        dic: {
          id: 1,
          code: "AE1",
          desc: "in transit"
        }
      };
    case 1:
      return {
        ...base,
        dic: {
          id: 2,
          code: "AS1",
          desc: "shipped"
        }
      };
    case 2:
      return {
        ...base,
        dic: {
          id: 3,
          code: "AS2",
          desc: "in transit"
        }
      };
    case 3:
      return {
        ...base,
        dic: {
          id: 4,
          code: "D6T",
          desc: "received"
        }
      };
    default:
      return {
        ...base,
        dic: {
          id: 5,
          code: "COR",
          desc: "confirmed"
        }
      };
  }
};

const createFakeApiDocuments = (count: number = 20): ApiDocument[] => {
  return Array.from({ length: count }).map(() => ({
    ...defaultApiDoc,
    statuses: Array.from({
      length: faker.random.number({ min: 1, max: 5 })
    }).map((_, idx) => idxToStatus(idx)),
    service_request: {
      id: faker.random.number(50),
      service_request: faker.random.alphaNumeric(8).toUpperCase()
    },
    part: {
      ...defaultApiDoc.part,
      nomen: faker.commerce.productName()
    },
    suppadd: {
      ...defaultApiDoc.suppadd,
      desc: faker.random.arrayElement([
        "Motor Transportation",
        "Armory",
        "Communications",
        "Electronics Maintenence"
      ])
    },
    addresses: defaultApiDoc.addresses.map(addy => ({
      ...addy,
      name: faker.company.companyName()
    }))
  }));
};

export default createFakeApiDocuments;
