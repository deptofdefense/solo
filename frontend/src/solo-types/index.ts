import { SortingRule, Filters } from "react-table";
export {
  default as createFakeApiDocs,
  defaultApiDoc,
  defaultApiResponse
} from "./apiDoc";
export * from "./warehouseUsers";
export { default as createFakeDocs, defaultDoc } from "./doc";

export interface Part {
  id: number;
  nsn: string;
  nomen: string;
  uom: string;
  price?: number;
  sac?: number;
  serial_control_flag?: string;
  lot_control_flag?: string;
  recoverability_code?: string;
  shelf_life_code?: number;
  controlled_inv_item_code?: string;
}

export interface Dic {
  id: number;
  code: string;
  desc: string;
}

export interface Status {
  id: number;
  dic: Dic;
  status_date: string;
  key_and_transmit_date?: string | null;
  esd?: string | null;
  received_qty: number | null;
  projected_qty: number;
  document: number;
  user?: number | null;
}

export interface AddressType {
  id: number;
  type: string;
  desc?: string | null;
}

export interface Address {
  id: number;
  address_type: AddressType;
  name: string;
  ric: string;
  addy1: string;
  addy2: string;
  addy3: string;
  city: string;
  state: string;
  zip: string;
  country: string;
  document: number[];
}

export interface ServiceRequest {
  id: number;
  service_request: string;
}

export interface Locator {
  id: number;
  code: string;
  desc?: string;
  subinventorys?: number;
}

export interface Subinventory {
  id: number;
  locators: Locator[];
  code: string;
  desc?: string;
  suppadd?: number;
}

export type LocatorMap = Record<string, Locator[]>;

export interface SuppAdd {
  id: number;
  code: string;
  subinventorys: Subinventory[];
}

interface BaseDocument {
  id: number;
  sdn: string;
  part: Part;
  statuses: Status[];
}

export interface ApiDocument extends BaseDocument {
  service_request: ServiceRequest;
  addresses: Address[];
  suppadd: SuppAdd;
}

export interface Document extends BaseDocument {
  serviceRequest: ServiceRequest;
  shipper?: Address;
  receiver?: Address;
  loadingStatus: LoadingStatus;
  subinventorys: Omit<Subinventory, "locators">[];
  locatorsBySubinventory: LocatorMap;
  commodityName: string;
  mostRecentStatusIdx: number;
  enteredReceivedQty: number;
  enteredReceivedBy: string;
  enteredSubinventoryCode: string;
  enteredLocatorCode: string;
}

export interface LoadingStatus {
  loading: boolean;
  error?: boolean;
  message?: string;
}

export interface Query<T extends object> {
  sort: SortingRule<T>[];
  page: number;
  filters: Filters<T>;
}

export interface PaginatedApiResponse<T> {
  results: T;
  count: number;
  next: number;
  previous: number;
}

export interface WarehouseUser extends LoadingStatus {
  userId: number;
  username: string;
  canD6T: boolean;
  canCOR: boolean;
  aac: string;
  hasModified?: boolean;
}
