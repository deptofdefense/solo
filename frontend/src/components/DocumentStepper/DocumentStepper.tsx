import React, { useMemo } from "react";
import classes from "./DocumentStepper.module.scss";
import { Status } from "solo-types";
import Step from "./DocumentStepperStep";

interface DocumentStepperProps {
  statuses: Status[];
}

const DocumentStepper: React.FC<DocumentStepperProps> = ({ statuses }) => {
  const statusByDic: Record<string, string> = useMemo(
    () =>
      statuses.reduce(
        (allDics, newStat) => ({
          ...allDics,
          [newStat.dic.code]: newStat.status_date
        }),
        {}
      ),
    [statuses]
  );

  return (
    <div className={classes.stepRoot}>
      <Step
        first
        title="Requested (AE1)"
        occuredAt={statusByDic["AE1"]}
        complete={statusByDic["AE1"]}
      />

      <Step
        title="Shipped (AS1)"
        occuredAt={statusByDic["AS1"]}
        complete={statusByDic["AS1"]}
      />

      <Step
        title="In Transit"
        occuredAt={statusByDic["AS2"]}
        complete={statusByDic["AS2"]}
      />

      <Step
        title="Received (D6T)"
        occuredAt={statusByDic["D6T"]}
        complete={statusByDic["D6T"]}
      />

      <Step
        title="Confirmation (COR)"
        occuredAt={statusByDic["D6T"]}
        complete={statusByDic["D6T"]}
        last
      />
    </div>
  );
};

export default DocumentStepper;
