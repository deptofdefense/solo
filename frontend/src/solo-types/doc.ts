import { Document } from "solo-types";

export const defaultDoc: Document = {
  id: 1,
  statuses: [
    {
      id: 1,
      dic: {
        id: 1,
        code: "AS1",
        desc: ""
      },
      status_date: "2020-03-01T21:47:13-05:00",
      key_and_transmit_date: null,
      esd: "2020-03-20",
      projected_qty: 2,
      received_qty: null,
      document: 1
    },
    {
      id: 2,
      dic: {
        id: 2,
        code: "AB1",
        desc: ""
      },
      status_date: "2020-03-05T21:47:56-05:00",
      key_and_transmit_date: null,
      esd: null,
      projected_qty: 2,
      received_qty: null,
      document: 1
    }
  ],
  part: {
    id: 2,
    nsn: "5430015061999",
    nomen: "TANK,LIQUID STORAGE",
    uom: "ea",
    price: 15,
    sac: 1,
    serial_control_flag: "n",
    lot_control_flag: "n",
    recoverability_code: "z",
    shelf_life_code: 0,
    controlled_inv_item_code: "u"
  },
  serviceRequest: {
    id: 1,
    service_request: "12345678"
  },
  suppadd: {
    id: 5,
    code: "YMTM",
    desc: "Motor Transportation",
    subinventorys: []
  },
  shipper: {
    id: 4,
    address_type: {
      id: 4,
      type: "Bill-To",
      desc: null
    },
    name: "AAC-M30300",
    ric: "SMS",
    addy1: "addy1",
    addy2: "addy2",
    addy3: "bldg 24006 Montezuma",
    city: "Arlington",
    state: "VA",
    zip: "22202",
    country: "United States",
    document: [1]
  },
  receiver: {
    id: 3,
    address_type: {
      id: 3,
      type: "Requester",
      desc: null
    },
    name: "requestor",
    ric: "5E3",
    addy1: "addy1",
    addy2: "addy2",
    addy3: "addy3",
    city: "Arlington",
    state: "VA",
    zip: "22202",
    country: "United States",
    document: [1]
  },
  sdn: "M3030012341234",
  commodityName: "Armory",
  loadingStatus: {
    loading: false
  },
  mostRecentStatusIdx: 1,
  enteredReceivedBy: "",
  enteredReceivedQty: 2
};

const createFakeDocuments = (count: number = 20): Document[] => {
  return Array.from({ length: count }).map(() => ({
    ...defaultDoc,
    id: Math.floor(Math.random() * 100)
  }));
};

export default createFakeDocuments;
