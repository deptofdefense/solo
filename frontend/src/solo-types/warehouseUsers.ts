import { WarehouseUser, PaginatedApiResponse } from "solo-types";
import * as faker from "faker";

export const defaultWareHouseUser: WarehouseUser = {
  userId: 1,
  username: "1234123123",
  first_name: "testFirst",
  last_name: "testLast",
  aac: "testaac",
  canD6T: false,
  canCOR: false,
  loading: false
};

export const defaultUserApiResponse: PaginatedApiResponse<WarehouseUser[]> = {
  results: [defaultWareHouseUser],
  count: 10,
  next: 2,
  previous: 1
};

export const createFakeUsers = (count: number = 25): WarehouseUser[] => {
  return Array.from({ length: count }).map((_, idx) => ({
    ...defaultWareHouseUser,
    userId: idx + 1,
    username: faker.finance.account(10),
    first_name: faker.name.firstName(),
    last_name: faker.name.lastName(),
    aac: faker.random.arrayElement(["M30300", "M30305"]),
    canCOR: faker.random.boolean(),
    canD6T: faker.random.boolean()
  }));
};
