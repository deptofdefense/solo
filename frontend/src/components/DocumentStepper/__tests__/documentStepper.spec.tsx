import React from "react";
import DocumentStepper from "../DocumentStepper";
import DocumentStepperStep from "../DocumentStepperStep";
import { defaultDoc } from "solo-types";
import { render } from "test-utils";

jest.mock("date-fns", () => ({
  parseISO: () => "testiso",
  formatDistanceToNow: () => {
    return "some amount of time";
  }
}));

afterAll(() => {
  jest.restoreAllMocks();
});

describe("DocumentStepperStep component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(
      <DocumentStepperStep title="teststep" complete />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("does not include the left bar if it's the first element", () => {
    const { container } = render(
      <DocumentStepperStep first title="teststep" />
    );
    expect(container.querySelector("div.stepRight")).toBeInTheDocument();
    expect(container.querySelector("div.stepLeft")).not.toBeInTheDocument();
  });

  it("does not include the right bar if it's the last element", () => {
    const { container } = render(<DocumentStepperStep last title="teststep" />);
    expect(container.querySelector("div.stepRight")).not.toBeInTheDocument();
    expect(container.querySelector("div.stepLeft")).toBeInTheDocument();
  });

  it("has green color if completed", () => {
    const { container } = render(
      <DocumentStepperStep title="teststep" complete />
    );
    expect(container.firstChild).toHaveClass("stepComplete");
  });

  it("has grey color if not complete", () => {
    const { container } = render(
      <DocumentStepperStep title="teststep" complete={false} />
    );
    expect(container.firstChild).toHaveClass("stepIncomplete");
  });

  it("includes formatted date if included", () => {
    const { queryByText } = render(
      <DocumentStepperStep title="teststep" occuredAt="somedate" />
    );
    expect(queryByText("some amount of time ago")).toBeInTheDocument();
  });

  it("uses dot circle icon when incomplete", () => {
    const { container } = render(
      <DocumentStepperStep title="teststep" complete={false} />
    );
    expect(
      container.querySelector("svg[data-icon='dot-circle']")
    ).toBeInTheDocument();
  });

  it("uses dot circle icon when incomplete", () => {
    const { container } = render(
      <DocumentStepperStep title="teststep" complete />
    );
    expect(
      container.querySelector("svg[data-icon='check-circle']")
    ).toBeInTheDocument();
  });
});

describe("DocumentStepper component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(
      <DocumentStepper statuses={defaultDoc.statuses} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
