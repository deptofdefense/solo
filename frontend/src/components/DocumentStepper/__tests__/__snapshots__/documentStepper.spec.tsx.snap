// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentStepper component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="stepRoot"
  >
    <div
      class="step stepIncomplete"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-dot-circle fa-w-16 stepIcon"
        data-icon="dot-circle"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm80 248c0 44.112-35.888 80-80 80s-80-35.888-80-80 35.888-80 80-80 80 35.888 80 80z"
          fill="currentColor"
        />
      </svg>
      <div>
        Requested (AE1)
      </div>
      <div
        class="stepRight"
      />
    </div>
    <div
      class="step stepComplete"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-check-circle fa-w-16 stepIcon"
        data-icon="check-circle"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
          fill="currentColor"
        />
      </svg>
      <div>
        Shipped (AS1)
      </div>
      <div>
        some amount of time ago 
      </div>
      <div
        class="stepLeft"
      />
      <div
        class="stepRight"
      />
    </div>
    <div
      class="step stepIncomplete"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-dot-circle fa-w-16 stepIcon"
        data-icon="dot-circle"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm80 248c0 44.112-35.888 80-80 80s-80-35.888-80-80 35.888-80 80-80 80 35.888 80 80z"
          fill="currentColor"
        />
      </svg>
      <div>
        In Transit
      </div>
      <div
        class="stepLeft"
      />
      <div
        class="stepRight"
      />
    </div>
    <div
      class="step stepIncomplete"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-dot-circle fa-w-16 stepIcon"
        data-icon="dot-circle"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm80 248c0 44.112-35.888 80-80 80s-80-35.888-80-80 35.888-80 80-80 80 35.888 80 80z"
          fill="currentColor"
        />
      </svg>
      <div>
        Received (D6T)
      </div>
      <div
        class="stepLeft"
      />
      <div
        class="stepRight"
      />
    </div>
    <div
      class="step stepIncomplete"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-dot-circle fa-w-16 stepIcon"
        data-icon="dot-circle"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm80 248c0 44.112-35.888 80-80 80s-80-35.888-80-80 35.888-80 80-80 80 35.888 80 80z"
          fill="currentColor"
        />
      </svg>
      <div>
        Confirmation (COR)
      </div>
      <div
        class="stepLeft"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`DocumentStepperStep component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="step stepComplete"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-check-circle fa-w-16 stepIcon"
      data-icon="check-circle"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
        fill="currentColor"
      />
    </svg>
    <div>
      teststep
    </div>
    <div
      class="stepLeft"
    />
    <div
      class="stepRight"
    />
  </div>
</DocumentFragment>
`;
